crossdock/client/echo.go

```diff
@@ -0,0 +1,81 @@
+package client
+
+import (
+	"crypto/rand"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"encoding/base64"
+
+	"golang.org/x/net/context"
+
+	"github.com/yarpc/yarpc-go"
+	"github.com/yarpc/yarpc-go/encoding/json"
+	"github.com/yarpc/yarpc-go/transport"
+	"github.com/yarpc/yarpc-go/transport/http"
+)
+
+// EchoRequest contains a message to echo
+type EchoRequest struct {
+	Token string `json:"token"`
+}
+
+// EchoResponse contains a messaged echoed by EchoRequest
+type EchoResponse struct {
+	Token string `json:"token"`
+}
+
+// EchoBehavior asserts that a server response is the same as the request
+func EchoBehavior(server string) Result {
+
+	yarpc := yarpc.New(yarpc.Config{
+		Name: "client",
+		Outbounds: transport.Outbounds{
+			"yarpc-test": http.Outbound(fmt.Sprintf("http://%v:8081", server)),
+		},
+	})
+	client := json.New(yarpc.Channel("yarpc-test"))
+	ctx := context.Background()
+
+	var response EchoResponse
+	token := randString(5)
+
+	_, err := client.Call(
+		ctx,
```

Comment:
For each call, you're supposed to create a child context. In this case, You can fix this later, but add a TODO here in that case.